internal/devconfig/configfile/packages.go

```diff
@@ -366,3 +368,134 @@ func packagesFromLegacyList(packages []string) []Package {
 	}
 	return packagesList
 }
+
+// PackageSpec specifies a Devbox package to install. Devbox supports a number
+// of package spec syntaxes:
+//
+//	| Syntax           | Example                     | Description                          |
+//	| ---------------- | --------------------------- | ------------------------------------ |
+//	| name@version     | go@v1.22                    | Resolved w/ search service           |
+//	| name             | go                          | Same as above; implies @latest       |
+//	| flake            | github:/nixos/nixpkgs#go    | Matches Nix installable syntax       |
+//	| attr path        | go                          | Equivalent to flake:nixpkgs#go       |
+//	| legacy attr path | go                          | Uses deprecated nixpkgs.commit field |
+//	| runx             | runx:golangci/golangci-lint | Experimental                         |
+//
+// Most package specs are ambiguous, making it impossible to tell which exact
+// syntax the user intended. PackageSpec parses as many as it can and leaves it
+// up to the caller to prioritize and resolve them.
```

Comment:
I think we should understand a string that has no version and is not a flake/runx to always mean attribute path. Thoughts?

Reply:
The tricky part is that almost any string is a valid flake reference due to indirect flakes. Still, I’d be onboard with saying we just don’t support scheme-less indirect flakes. So whereas Nix does:

```
foo → flake:foo
flake:foo → flake:foo
```

Devbox would do:

```
foo → flake:nixpkgs#foo
flake:foo → flake:foo
```

That still leaves ambiguity between Devbox packages and attribute paths though. We can't tell if `nodePackages.@angular/cli` is named `nodePackages.@angular/cli` or named `nodePackages.` with version `angular/cli` without querying the search service.